### ISSUE TITLE
Service shouldn't have spicy_ prefix for analyzers written in spicy.

### ISSUE THREAD

**Issue by JustinAzoff** [zeek/zeek#3289]
With more protocols being implemented in spicy, as well as existing ones being ported over it seems that we messed up a bit by having the service field show up as "spicy_x".

An Analyzer being written in spicy is an implementation detail.  Analyzers written in binpac did not show up as "binpac_x" and ones written in spicy shouldn't have a prefix either.  If we port an analyzer like ssh to spicy, the service field shouldn't change from "ssh" to "spicy_ssh".

**Comment by awelzel** [comment]
Can you see in how far or if this covers it?

https://github.com/zeek/package-template/issues/11

**Comment by JustinAzoff** [comment]
ah, so it's the individual analyzers that are doing that? I made this change to the ldap one and that seemed to work:

```
-protocol analyzer spicy::LDAP_TCP over TCP:
+protocol analyzer LDAP_TCP over TCP:
-protocol analyzer spicy::LDAP_UDP over UDP:
+protocol analyzer LDAP_UDP over UDP:
```

and that switched service from `spicy_ldap_tcp` to `ldap_tcp`.  I did note that I can't further change it to

```
protocol analyzer LDAP over TCP:
protocol analyzer LDAP over UDP:
```

for both, that spits out

```
fatal error: redefinition of protocol analyzer LDAP
```

but that may be a separate issue.

So, I'm not sure what should be done here.  In the longer term we can do some outreach / pull requests to update the existing spicy analyzers, but I wonder if in the short term if it would be a bad idea for zeek to strip the `spicy_` prefix.

**Comment by rsmmr** [comment]
> ah, so it's the individual analyzers that are doing that?

Yes, correct, that's just been a convention originally, nothing hardcoded.

> ```
> protocol analyzer LDAP over TCP:
> protocol analyzer LDAP over UDP:
> ```
 
> ```
> fatal error: redefinition of protocol analyzer LDAP
> ```
> 
> but that may be a separate issue.

It is, it's not possible currently to define two analyzers with the same name. For this particular case, there's also a pending feature request: https://github.com/zeek/spicy-plugin/issues/50. We should probably tackle that.

> So, I'm not sure what should be done here. In the longer term we can do some outreach / pull requests to update the existing spicy analyzers, but I wonder if in the short term if it would be a bad idea for zeek to strip the `spicy_` prefix.

What analyzers are we talking about here? Getting them fixed would certainly be the preferred solution, but I can see stripping the prefix Zeek-side for now if that's hard.

**Comment by JustinAzoff** [comment]
> What analyzers are we talking about here? Getting them fixed would certainly be the preferred solution, but I can see stripping the prefix Zeek-side for now if that's hard.


A github search finds most of them:

https://github.com/search?q=%22protocol+analyzer+spicy%3A%3A%22&type=code&ref=advsearch

I think just about all the common used ones have the prefix:

* ldap
* wireguard
* stun
* openvpn
* gena
* ssdp
* ipsec
* genisys

**Comment by bbannier** [comment]
> I think just about all the common used ones have the prefix:
>
> - ldap
> ...

We'll remove the `spicy_` prefix when moving spicy-ldap into Zeek proper with #3234.

All the other analyzers you have listed are outside of our control, and I do not think that we should after the fact and silently change names developers have picked deliberately (asking their developers to change the name is totally okay though).

In the packet template nothing generates a name containing `spicy_` anymore (see e.g., [here](https://github.com/zeek/package-template/blob/master/features/spicy-protocol-analyzer/analyzer/%40ANALYZER_LOWER%40.evt%40ALT-one-unit%40#L9)). The [suggestion for the package name](https://github.com/zeek/package-template/blob/aa2d1dd47189d0817edf38753b95101be25455e9/__init__.py#L272-L274) has no effect on the name of the analyzer.

**Comment by bbannier** [comment]
Closed by zeek/package-template#11.
